packages/webapp/src/_app/eos/secret-publisher.ts

```diff
@@ -0,0 +1,217 @@
+    // TODO: real implementation
+    // fetch eden.keys_table( where account = each account item )
+
+    const mockedKeysLen = accounts.length;
+    const mockedKeys = [];
+    for (let i = 0; i < mockedKeysLen; i++) {
+        mockedKeys.push(
+            "EOS5wGY8RcG8PuGYTPU9QwmiEm9XReBkK4SGtSuS4f1oeX4TNGaHT"
+        );
+        // temp PK: 5KcMypBPGByXbeBphsVzyyzUvg31tWNKsgrfXs2MmanYgYf4gao
+    }
+    return mockedKeys;
+};
+
+const generateRandomTransientKey = async () => {
+    const privateKey = await ecc.randomKey();
+    const publicKey = ecc.privateToPublic(privateKey);
+    return { privateKey, publicKey };
+};
+
+const ecdhRecipientsKeyEncriptionKeys = async (
+    recipientPublicKeys: string[],
+    transientPrivateKeyString: string
+) => {
+    const transientPrivateKey = PrivateKey.fromString(
+        transientPrivateKeyString
+    );
+    const transientEc = ec.keyPair({
+        priv: new bn(transientPrivateKey.d.toHex(), 16),
+    });
+    return await Promise.all(
+        recipientPublicKeys.map(async (publicKey: string) => {
+            const recipientEcPublicKey = ec.keyFromPublic(
+                PublicKey.fromString(publicKey).toHex(),
+                "hex"
+            );
+            const ecdhSecret = transientEc.derive(
+                recipientEcPublicKey.getPublic()
+            );
+            return await hkdfSha256FromEcdh(
+                ecdhSecret.toArrayLike(ArrayBuffer) // TODO: review endianness, empty = Big Endian
+            );
+        })
+    );
+};
+
+const hkdfSha256FromEcdh = async (ecdhSecret: ArrayBuffer) => {
+    const sharedSecretKey = await crypto.subtle.importKey(
+        "raw",
+        ecdhSecret,
+        { name: "HKDF" },
+        false,
+        ["deriveKey", "deriveBits"]
+    );
+    const derivedKey = await crypto.subtle.deriveBits(
+        {
+            name: "HKDF",
+            hash: "SHA-256",
+            salt: new Uint8Array([]),
+            info: new Uint8Array([]),
+        },
+        sharedSecretKey,
+        256
+    );
+    return new Uint8Array(derivedKey);
+};
+
+const generateRandomSessionKey = async () => {
+    const key = await crypto.subtle.generateKey(
+        {
+            name: "AES-GCM",
+            length: 128,
+        },
+        true,
+        ["encrypt", "decrypt"]
+    );
+    return new Uint8Array(await crypto.subtle.exportKey("raw", key));
+};
+
+const encryptSessionKeys = async (
+    rawSessionKey: Uint8Array,
+    keks: Uint8Array[]
+) => {
+    const sessionKey = await crypto.subtle.importKey(
+        "raw",
+        rawSessionKey,
+        { name: "AES-GCM" },
+        true,
+        ["encrypt", "decrypt"]
+    );
+
+    return await Promise.all(
+        keks.map(async (rawKek: Uint8Array) => {
+            console.info("importing kek");
+            const kek = await crypto.subtle.importKey(
+                "raw",
+                rawKek,
+                { name: "HKDF" },
```

Comment:
Is this right?  kek should be an AES-128 key and have wrapKey as the keyUsage.

Reply:
fixed!